### PR TITLE
chore: Setup ktlint as a pre-commit hook

### LINKED ISSUE #56: Use a kotlin formatter
We currently do not have a strict code formatting discipline.
I propose we use [ktlint](https://github.com/pinterest/ktlint) as it's a well documented, well maintainted tool.

There is several option to enforce ktlint:

- Add a github action like https://github.com/ScaCap/action-ktlint which will check if the modified files pass validation -> only drawback is that we are not so sure about this action, if it will stay maintained. We might have to help?
- We can use one of the ktlint gradle plugin together with a Travis config that runs the check, like https://github.com/kafumi/android-danger-ktlint-sample/blob/master/.travis.yml (gradle plugins are documented on the ktlint repo, see https://github.com/pinterest/ktlint#-with-gradle)
- We can have a bot that once in a while runs ktlint on every file, to auto-format what can be formatted such as this script: https://gist.github.com/wasabeef/60113a9d4dec64ad71a901c089aae8b5
- We can write our own script to do our own stuff.

### PR DESCRIPTION
**Pull request by redwarp**
Adds a local pre-commit hook to bento to pass kotlin file through ktlint format. Inspired by https://github.com/maltzj/ktlint-pre-commit

Also ran `pre-commit autoupdate`, which updated the yelp-secret hook.

When running the pre-commit hook without a change to kotlin files:

```
▶ git amend
Detect secrets...........................................................Passed
Ktlint format for your kotlin code...................(no files to check)Skipped
[bvermont/56/setup_ktlint bb75295] chore: Setup ktlint as a pre-commit hook
 Date: Thu Mar 12 14:38:00 2020 +0100
 4 files changed, 38 insertions(+), 10 deletions(-)
 rewrite .pre-commit-config.yaml (66%)
 create mode 100755 hooks/run-ktlint-format.sh
```

Running the hook on the whole project:

```
▶ pre-commit run --all-files
Detect secrets...........................................................Passed
Ktlint format for your kotlin code.......................................Failed
- hook id: ktlint-format
- files were modified by this hook

(…)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   bento-sample-app/src/androidTest/java/com/yelp/android/bentosampleapp/ListViewActivity_BentoInteractionTest.kt
	modified:   bento-sample-app/src/androidTest/java/com/yelp/android/bentosampleapp/MainActivityTest.kt
	modified:   bento-sample-app/src/androidTest/java/com/yelp/android/bentosampleapp/RecyclerViewScrollTest.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/ComponentReplacementActivity.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/HorizontalGridComponentsActivity.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/ListViewActivity.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/ListVisibilityActivity.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/MainActivity.kt
	modified:   bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/RecyclerViewActivity.kt

(…)
```

### CODE REVIEW

**Comment by redwarp** [comment]
Let's discuss! I'm okay with using a different linter/formatter.
Ktlint is easy to setup, and does some of the basic things like "removing extra white lines", adding missing white space, white space end of file etc...

I don't know if this is sufficient, but will still solve some of the basic white space or no white space discussions.

For example, ktlint don't seem to have an opinion between
```
private lateinit var componentController: RecyclerViewComponentController
```

and
```
private lateinit var componentController:
        RecyclerViewComponentController
```

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+repos:
+  - repo: git@github.com:Yelp/detect-secrets
+    rev: v0.13.0
     hooks:
-    -   id: detect-secrets
-        args: ['--baseline', '.secrets.baseline']
+      - id: detect-secrets
+        args: ["--baseline", ".secrets.baseline"]
         exclude: .*/build/.*|.*/test/.*|.*/androidTest/.*
+  - repo: local
```
Comment (cortinico) [inline .pre-commit-config.yaml]: You can actually use @macisamuele 's KtLint wrapper:

```
- repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
  rev: v1.3.2
  hooks:
  - id: pretty-format-kotlin
    args: [--autofix]
```

As we did for `swagger-gradle-codegen`:

https://github.com/Yelp/swagger-gradle-codegen/blob/master/.pre-commit-config.yaml#L14-L20

Comment (redwarp) [reply .pre-commit-config.yaml]: Where is the fun in that? (Though, good idea)

**Comment by redwarp** [comment]
Updating to use @cortinico 's suggestion

**Comment by redwarp** [comment]
Removed unnecessary change to GlobalDependencies.kt (an artifact of the pre-commit hook)

**Comment by redwarp** [comment]
And remove script, no longer being necessary.

**Review by dwaxemberg — APPROVED** [review]

**Pull request merged**